src/openvic-simulation/map/ProvinceInstance.cpp

```diff
@@ -185,6 +191,73 @@ void ProvinceInstance::_update_pops(DefineManager const& define_manager) {
 	}
 }
 
+void ProvinceInstance::update_modifier_sum(Date today, StaticModifierCache const& static_modifier_cache) {
+	// Update sum of province modifiers
+	base_modifier_sum.clear();
+
+	// Erase expired event modifiers and add non-expired ones to the sum
+	std::erase_if(event_modifiers, [this, today](ModifierInstance const& modifier) -> bool {
+		if (today <= modifier.get_expiry_date()) {
+			base_modifier_sum.add_modifier(*modifier.get_modifier(), this);
+			return false;
+		} else {
+			return true;
+		}
+	});
+
+	// Add static modifiers
+	if (static_modifier_cache.get_core() != nullptr && is_owner_core()) {
+		base_modifier_sum.add_modifier(*static_modifier_cache.get_core(), this);
+	}
+	if (province_definition.is_water()) {
+		if (static_modifier_cache.get_sea_zone() != nullptr) {
+			base_modifier_sum.add_modifier(*static_modifier_cache.get_sea_zone(), this);
+		}
+	} else {
+		if (static_modifier_cache.get_land_province() != nullptr) {
+			base_modifier_sum.add_modifier(*static_modifier_cache.get_land_province(), this);
+		}
+
+		if (province_definition.is_coastal()) {
+			if (static_modifier_cache.get_coastal() != nullptr) {
+				base_modifier_sum.add_modifier(*static_modifier_cache.get_coastal(), this);
+			}
+		} else {
+			if (static_modifier_cache.get_non_coastal() != nullptr) {
+				base_modifier_sum.add_modifier(*static_modifier_cache.get_non_coastal(), this);
+			}
+		}
+
+		// TODO - overseas, blockaded, no_adjacent_controlled, has_siege, occupied, nationalism, infrastructure
+	}
+
+	for (BuildingInstance const& building : buildings.get_items()) {
+		base_modifier_sum.add_modifier(building.get_building_type(), this);
+	}
+
+	if (crime != nullptr) {
+		base_modifier_sum.add_modifier(*crime, this);
+	}
+
+	if (province_definition.get_continent() != nullptr) {
+		base_modifier_sum.add_modifier(*province_definition.get_continent(), this);
+	}
+
+	if (province_definition.get_climate() != nullptr) {
```

Comment:
Can climate even be null?

Reply:
Same as above, if a province isn't present in any climate's list of provinces then it will have `nullptr` for its climate.